transformations/numeric_to_word/numeric2word.py

```diff
@@ -0,0 +1,1025 @@
+    return words
+
+def numeric_beside_end_bracket_to_words(x):
+    end_digit_index = len(x) - re.search(r"\d", x[::-1]).start()
+    words = recognize_transform(x[:end_digit_index], ' ', ' ') + ' ' + x[end_digit_index:]
+    return words
+
+def natural_log_to_words(x):
+    end_digit_index = len(x) - re.search(r"\d", x[::-1]).start()
+    words = 'e ( ' + recognize_transform(x[2:end_digit_index], ' ', ' ') + ' ' + x[end_digit_index:]
+    return words
+
+def fraction_to_words(x):
+    divider_index = x.find('/')
+    
+    numerator = x[:divider_index]
+    denominator = x[divider_index+1:]
+    
+    if numerator == '1' and denominator == '2':
+        words = 'a half'
+    elif numerator == '1' and denominator == '3':
+        words = 'a third'
+    elif numerator == '2' and denominator == '3':
+        words = 'two third'
+    elif numerator == '1' and denominator == '4':
+        words = 'a quarter'
+    elif numerator == '2' and denominator == '4':
+        words = 'two quarter'
+    elif numerator == '3' and denominator == '4':
+        words = 'three quarter'
+    else:
+        words = num2words(numerator) + ' over ' + num2words(denominator)
+    return words
+
+### Supplements
+
+class TimeInWords2():
+    def __init__(self):
+        self.words=["one", "two", "three", "four", "five", "six", "seven", "eight","nine", 
+       "ten", "eleven", "twelve", "thirteen", "fourteen", "quarter", "sixteen",
+       "seventeen", "eighteen", "nineteen", "twenty", "twenty one", 
+       "twenty two", "twenty three", "twenty four", "twenty five", 
+       "twenty six", "twenty seven", "twenty eight", "twenty nine", "half"]
+         
+    def caltime(self, time_string):
+        splitter = ''.join([i for i in time_string if not i.isdigit()])
+        hrs = int(time_string.split(splitter)[0])
+        mins = int(time_string.split(splitter)[1])
+        header = "" #It is..
+        msg=""
+        am_pm = "am"
+        if (hrs >12):
+            hrs=hrs-12
+            am_pm = "pm"
+        if (mins == 0):
+            hr = self.words[hrs-1]
+            msg=header + hr + " o'clock" + " " + am_pm #+ "."
+        elif (mins < 31):      
+               hr = self.words[hrs-1]
+               mn = self.words[mins-1]
+               msg = header + mn + " past " + hr + " " + am_pm #+ "."
+        else:
+            hr = self.words[hrs]
+            mn =self.words[(60 - mins-1)]
+            msg = header + mn + " to " + hr + " " + am_pm #+ "."
+        return msg
+    
+def m_bn_currency_to_words(x):
+    end_digit_index = len(x) - re.search(r"\d", x[::-1]).start()
+    
+    word_list = recognize_transform(x[:end_digit_index], '', '').split()
+    
+    m_or_bn = x[end_digit_index:].split(',')[0]
+    sticky_following = x[end_digit_index:].split(',')[1:]
+    currency = [word_list[-1]]
+    
+    words = ' '.join(word_list[:-1] + [m_or_bn] + currency + sticky_following)
+    return words
+    
+symbol_to_currency_name_dict = {"؋":"Afghanistani Afghani", 
```

Comment:
Just for clean code, create a constant file and move all the constants to that file instead of putting it in a single file.

Reply:
done